packages/framework/fluid-static/src/compatibilityConfiguration.ts

```diff
@@ -44,7 +43,8 @@ export const compatibilityModeRuntimeOptions: Record<
 		// The runtime ID compressor is a prerequisite to use SharedTree but is off by default and must be explicitly enabled.
 		// It introduces a new type of op which is not compatible with 1.x clients.
 		enableRuntimeIdCompressor: "on",
-		// GC is not yet enabled by default, although it could be enabled here from a compat standpoint
-		gcOptions: { enableGCSweep: undefined, gcDisableThrowOnTombstoneLoad: true },
+		// Explicitly disable running Sweep for 2.x clients. Although sweep is supported for 2.x, it will break if 1.x and 2.x
+		// clients are running in parallel.
```

Comment:
This comment is incorrect.  So is the one above.  "1" section means, when running _this code_ (which is 2.x) in an environment where 1.x clients are collaborating, use these.  The "2" section means, when running this code where _all collaborators are also on 2.x_, use these settings.

Reply:
I think the reason we're not enabling it by default here is that it's not enabled by default in the GC code itself.  This is just extra explicit.  Consider that ODSP Client also will have GC disabled, because of GC settings.

Whenever we enable GC by default, we can remove this.  Or we could remove it now and put a comment saying "In 2.0 GC Sweep is still disabled by default. In a later minor version it will be enabled".  And we can open an ADO item for making that switch.

And... Don't we have enough prod experience under our belt to enable it now?  What are we waiting for?

Reply:
Oh, I compeltely missed that. Thanks for pointing it out. I updated the comments to reflect this.

> And... Don't we have enough prod experience under our belt to enable it now? What are we waiting for?

I believe we can enable it. My understanding is that for 2.0, we are being careful about features we ship and evaluating risk-benefit. In case of sweep, there is no benefit because there is only one data store (root) and there are no attachment blobs. So, there is only risk.